usr/src/uts/common/fs/zfs/sys/uberblock_impl.h

```diff
@@ -54,6 +55,12 @@ struct uberblock {
 
 	/* highest SPA_VERSION supported by software that wrote this txg */
 	uint64_t	ub_software_version;
+
+	/* These fields are reserved for features that are under development: */
+	uint64_t	ub_mmp_magic;
+	uint64_t	ub_mmp_delay;
```

Comment:
The uberblock is padded out with zeros to (at least) 512 bytes.  So this change doesn't actually change anything about the on-disk state (since the new fields are not set, in this change).  The fields continue to be zero, and we continue to ignore them.  This change is just for humans to know that they shouldn't use these offsets for other purposes.

Also, the test suite opens older pools (that were generated with ancient versions of ZFS and checked into the source tree), so we know it works.

Questions as to how the new fields will be used when their respective features (MMP, checkpoint) are integrated are best directed to those reviews.  But they obviously need to deal with these fields being zero when dealing with older pools.  Side note: older software will ignore these fields, so backwards compatibility with these features is managed with feature flags, not the uberblock.